sqlalchemy_ibmi/base.py

```diff
@@ -16,7 +16,104 @@
+
+Connection Arguments
+-----------------
+
+The sqlalchemy-ibmi dialect supports multiple connection arguments that are
+passed to the URL in create_engine function.
+
+* ``autocommit`` - If False, Connection.commit must be called;
+  otherwise each statement is automatically committed.
+  Defaults to ``False``.
+* ``readonly`` - defaults to ``False``.
+  If True, the connection is set to readonly
+* ``timeout`` - A timeout for the connection, in seconds.
+  This causes the connection's SQL_ATTR_LOGIN_TIMEOUT to be
+  set before the connection occurs.
+
+Transaction Isolation Level / Autocommit
+----------------------------------------
+Db2 for i supports 5 isolation levels:
+
+* ``SERIALIZABLE``: ``*RR``
+* ``READ COMMITTED``: ``*CS``
+* ``READ UNCOMMITTED``: ``*CHG``
+* ``REPEATABLE READ``: ``*ALL``
+* ``*`` ``NO COMMIT``: ``*NONE/*NC``
+
+**At this time, pyODBC and SQLAlchemy support all of these isolation levels
+except NO COMMIT.**
+
+Autocommit is supported in Db2 for i by passing the TRUEAUTOCOMMIT value in the
+connection arguments. Autocommit is supported on all available isolation levels.
+
+To set isolation globally::
+
+    engine = create_engine("ibmi://user:pass@host/",
+    isolation_level='REPEATABLE_READ')
+
+To set using per-connection execution options::
+
+    connection = engine.connect()
+    connection = connection.execution_options(
+        isolation_level="SERIALIZABLE"
+    )
+
+Table Creation String Size
+--------------------------
+When creating a table with sqlalchemy, Db2 for i requires that a string size
+is provided.
+
+Provide the length for in the CREATE TABLE as follows:
+
+.. code-block:: python
+   :emphasize-lines: 4, 8
+
+    class User(Base):
+        __tablename__ = 'users'
+        id = Column(Integer, Sequence('user_id_seq'), primary_key=True)
+        name = *Column(String(50))*
+
+    users = Table('users', metadata,
+        Column('id', Integer, Sequence('user_id_seq'), primary_key=True),
+        Column('name', String(50)),
+    )
+
+
+Query Function Strings
+----------------------
+Db2 for i doesn't support parameter markers in SELECT statments, so as a
+result the following command will not work with sqlalchemy-ibmi::
+
+    session.query(func.count("*")).select_from(User).scalar()
+
+Instead, please use the following::
+
+    session.query(func.count()).select_from(User).scalar()
+
+Addtionally, column names cannot be passed as strings, so convert your column
+strings to literal columns as follows::
+
+    from sqlalchemy.sql import literal_column
+    session.query(func.count(literal_column("colname"))).select_from(User).scalar()
+
+Text search support
+-------------------
+The ColumnOperators.match function can be used with Db2 for i using the
+CONTAINS function. However, it requires that OmniFind Text Search Server for
+Db2® for i be installed and started to use this. If the Text Search Server is
+not installed, then a LIKE operation is completed.
```

Comment:
```suggestion
The ColumnOperators.match function is implemented using a basic LIKE operation
by default. However, when OmniFind Text Search Server for Db2 for i is
installed, match will take advantage of the CONTAINS function that it provides.
```

We should also make "OmniFind Text Search Server for Db2 for i" a link to https://www.ibm.com/support/knowledgecenter/ssw_ibm_i_74/rzash/rzashkickoff.htm